core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java

```diff
@@ -0,0 +1,95 @@
+package org.stellar.anchor.sep6;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.stellar.anchor.api.sep.AssetInfo;
+import org.stellar.anchor.api.sep.sep6.InfoResponse;
+import org.stellar.anchor.api.sep.sep6.InfoResponse.*;
+import org.stellar.anchor.asset.AssetService;
+import org.stellar.anchor.config.Sep6Config;
+
+public class Sep6Service {
+  private final Sep6Config sep6Config;
+  private final AssetService assetService;
+
+  private final InfoResponse infoResponse;
+
+  public Sep6Service(Sep6Config sep6Config, AssetService assetService) {
+    this.sep6Config = sep6Config;
+    this.assetService = assetService;
+    this.infoResponse = buildInfoResponse();
+  }
+
+  public InfoResponse getInfo() {
+    return infoResponse;
+  }
+
+  private InfoResponse buildInfoResponse() {
+    InfoResponse response =
+        InfoResponse.builder()
+            .deposit(new HashMap<>())
+            .depositExchange(new HashMap<>())
+            .withdraw(new HashMap<>())
+            .withdrawExchange(new HashMap<>())
+            .fee(
+                FeeResponse.builder()
+                    .enabled(false)
+                    .description("Fee endpoint is not supported.")
+                    .build())
+            .transactions(
+                TransactionsResponse.builder().enabled(true).authenticationRequired(true).build())
+            .transaction(
+                TransactionResponse.builder().enabled(true).authenticationRequired(true).build())
+            .features(
+                FeaturesResponse.builder()
+                    .accountCreation(sep6Config.getFeatures().isAccountCreation())
+                    .claimableBalances(sep6Config.getFeatures().isClaimableBalances())
+                    .build())
+            .build();
+
+    for (AssetInfo asset : assetService.listAllAssets()) {
+      if (asset.getSep6Enabled()) {
+
+        if (asset.getDeposit().getEnabled()) {
+          List<String> methods = asset.getDeposit().getMethods();
+          AssetInfo.Field type =
+              AssetInfo.Field.builder()
+                  .description("type of deposit to make")
+                  .choices(methods)
+                  .build();
+
+          DepositAssetResponse deposit =
+              DepositAssetResponse.builder()
+                  .enabled(true)
+                  .authenticationRequired(true)
+                  .fields(ImmutableMap.of("type", type))
+                  .build();
```

Comment:
`type` is optional in the spec today, should we be checking if `methods` is empty and omitting the `type` field if so? Alternatively, we should have validation that there is always at least one type provided in the config.

Reply:
> Alternatively, we should have validation that there is always at least one type provided in the config.

We took this approach. If SEP-6 and deposit/withdraw is enabled, AP will require that at least one method is specified. The validation is done as part of the `AssetService` initialization. https://github.com/stellar/java-stellar-anchor-sdk/blob/96d215de79e3acc51fb65b691a2e376b57e51aa3/core/src/test/kotlin/org/stellar/anchor/asset/DefaultAssetServiceTest.kt#L73-L99